numba/core/interpreter.py

```diff
@@ -540,14 +540,69 @@ def _remove_unused_temporaries(self):
         current block.
         """
         new_body = []
+        replaced_var = {}
         for inst in self.current_block.body:
-            if (isinstance(inst, ir.Assign)
-                    and inst.target.is_temp
-                    and inst.target.name in self.assigner.unused_dests):
-                continue
+            # the same temporary is assigned to multiple variables in cases
+            # like a = b[i] = 1, so need to handle replaced temporaries in
+            # later setitem/setattr nodes
+            if (isinstance(inst, (ir.SetItem, ir.SetAttr))
+                    and inst.value.name in replaced_var):
+                inst.value = replaced_var[inst.value.name]
```

Comment:
I found these cases since some existing tests include these corner cases. Added tests in https://github.com/numba/numba/pull/6575/commits/4aec70cc6c540cc59f456384c7aa273c9a11e958 to be sure.

Reply:
hmmm, I ran the whole test suite and could find anything that hit it, perhaps I was mistaken, anyway, thanks for adding the test.